File: two_point/151.反转字符串中的单词.go
/*
 * @lc app=leetcode.cn id=151 lang=golang
 *
 * [151] 反转字符串中的单词
 */

// @lc code=start

import (
	"strings"
)

func reverseWords(s string) string {
	res := make([]string, 0)
	splitStr := strings.Split(s, " ")
	for _, v := range splitStr {
		if v == "" {
			continue
		}
		res = append(res, v)
	}
	return reverse(res)
}

func reverse(s []string) string {
	left := 0
	right := len(s) - 1
	for left < right {
		s[left], s[right] = s[right], s[left]
		left++
		right--
	}
	return strings.Join(s[:], " ")
}

// @lc code=end

File: two_point/151.反转字符串中的单词.py
#
# @lc app=leetcode.cn id=151 lang=python3
#
# [151] 反转字符串中的单词
#


# @lc code=start
class Solution:
    def reverseWords(self, s: str) -> str:
        words = s.split()
        # 翻转
        words = words[::-1]
        return " ".join(words)


# @lc code=end
